### PR TITLE
fix: Footer Responsiveness

### PR DESCRIPTION
**Pull request by abhishekrathi16**
### Fixes #<number>

Description: The footer designed earlier by me had problems for responsiveness and the code was breaking down when paired with the navbar. This pull request has code which uses mui-grids as well as react props and mapping and ensures that the code doesn't break down in different screen sizes. Complements the navbar design.

#### Checklist
- [x] I've followed the Contributing guidelines provided in the repository.
- [x] I've made the changes which were demanded in the linked issue.
- [x] I've tested my code on a Chromium based browser.
- [x] I've tested my code on Mozilla Firefox.
- [x] My code gave a clean console on debugging. (no warnings/errors)

#### Testimonial
ss1
ss2
ss3

### CODE REVIEW

**Comment by abhishekrathi16** [comment]
Still need to add some proper sizing to the icons before it's good to go.

File: client/components/footer/Data.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import EmailIcon from "@mui/icons-material/Email";
+import LinkedInIcon from "@mui/icons-material/LinkedIn";
+import styles from "../../styles/Footer.module.css"
+
+
+const Data=[
+    {
+        id : 1,
+        title: "Quick Access",
+        line1: "B.Tech Course",
```
Comment (EmperorYP7) [inline client/components/footer/Data.js]: Why not make this an array?
```suggestion
        "quick_access": [
            {
                "line": "B.Tech Course",
                "link": ""
            },
            // .. and so on
        ],
```

File: client/components/footer/Data.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        link3: "",
+        link4: "",
+        link5: "",
+        icon: <div></div>,
+        icon2: <div></div>
+    },
+    {
+        id : 2,
+        title: "Student Corner",
+        line1: "Youtube Channel",
```
Comment (EmperorYP7) [inline client/components/footer/Data.js]: This as well.
```suggestion
        "student_corner": [
            {
                "line": "B.Tech Course",
                "link": ""
            },
            // .. and so on
        ],
```

File: client/components/footer/Data.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        link3: "",
+        link4: "",
+        link5: "",
+        icon: <div></div>,
+        icon2: <div></div>,
+    },
+    {
+        id : 3,
+        title: "Get In Touch",
+        line1: "tap@iiitranchi.ac.in",
```
Comment (EmperorYP7) [inline client/components/footer/Data.js]: Here as well

File: client/components/footer/Group.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
+import { Link, Box, Grid } from "@mui/material";
+import EmailIcon from "@mui/icons-material/Email";
+import LinkedInIcon from "@mui/icons-material/LinkedIn";
+import styles from "../../styles/Footer.module.css";
+
+export default function Group(props) {
+  return (
+    <>
+      <Grid
```
Comment (EmperorYP7) [inline client/components/footer/Group.js:9]: Sweet!

**Review by EmperorYP7 — CHANGES_REQUESTED** [review]

**Comment by abhishekrathi16** [comment]
The changes requested will not change the footer significantly although changing the data arrangement that I am passing as props will need me to change a lot in the code, perhaps the mapping and props as well as styling. Please re-consider the changes requested and if you still feel the need to change the code I'll do it without any issues. Also, update me about the footer design that you want for mobile view, I will update the code accordingly.

**Review by EmperorYP7 — CHANGES_REQUESTED** [review]
The output looks good to me. However, consider the case where we want to customize the fields in the footer; we'd need to restructure `Data.js` so that just by adding and removing elements from the list, we can produce correct results. So, better be it now.

**Comment by abhishekrathi16** [comment]
Okay, then I'll start working on it right away.
And do you want me to fix the mobile view as discussed in the group?

**Comment by EmperorYP7** [comment]
> And do you want me to fix the mobile view as discussed in the group?

Yes.

**Comment by abhishekrathi16** [comment]
The requested changes have been made, please review the changes and inform me if there is anything you would want me to do.
@EmperorYP7

**Review by EmperorYP7 — APPROVED** [review]
LGTM! well done.

**Pull request merged**